static/app/components/sidebar/sidebarDropdown/switchOrganization.tsx

```diff
@@ -21,67 +23,96 @@ type Props = {
-            data-test-id="sidebar-switch-org-menu"
-            {...getMenuProps({})}
+function SwitchOrganization({organizations, canCreateOrganization}: Props) {
+  const currentOrganization = useOrganization();
+
+  return (
+    <DeprecatedDropdownMenu isNestedDropdown>
+      {({isOpen, getMenuProps, getActorProps}) => (
+        <Fragment>
+          <SwitchOrganizationMenuActor
+            data-test-id="sidebar-switch-org"
+            {...getActorProps({})}
+            onClick={e => {
+              // This overwrites `DropdownMenu.getActorProps.onClick` which normally handles clicks on actor
+              // to toggle visibility of menu. Instead, do nothing because it is nested and we only want it
+              // to appear when hovered on. Will also stop menu from closing when clicked on (which seems to be common
+              // behavior);
+
+              // Stop propagation so that dropdown menu doesn't close here
+              e.stopPropagation();
+            }}
           >
-            <OrganizationList role="list">
-              {sortBy(organizations, ['status.id']).map(organization => {
-                const url = `/organizations/${organization.slug}/`;
-
-                return (
-                  <SidebarMenuItem key={organization.slug} to={url}>
-                    <SidebarOrgSummary organization={organization} />
-                  </SidebarMenuItem>
-                );
-              })}
-            </OrganizationList>
-            {organizations && !!organizations.length && canCreateOrganization && (
-              <Divider css={{marginTop: 0}} />
-            )}
-            {canCreateOrganization && (
-              <SidebarMenuItem
-                data-test-id="sidebar-create-org"
-                to="/organizations/new/"
-                style={{alignItems: 'center'}}
-              >
-                <MenuItemLabelWithIcon>
-                  <StyledIconAdd />
-                  <span>{t('Create a new organization')}</span>
-                </MenuItemLabelWithIcon>
-              </SidebarMenuItem>
-            )}
-          </SwitchOrganizationMenu>
-        )}
-      </Fragment>
-    )}
-  </DeprecatedDropdownMenu>
-);
+            {t('Switch organization')}
+
+            <SubMenuCaret>
+              <IconChevron size="xs" direction="right" />
+            </SubMenuCaret>
+          </SwitchOrganizationMenuActor>
+
+          {isOpen && (
+            <SwitchOrganizationMenu
+              data-test-id="sidebar-switch-org-menu"
+              {...getMenuProps({})}
+            >
+              <OrganizationList role="list">
+                {sortBy(organizations, ['status.id']).map(organization => {
+                  const {slug, links} = organization;
+                  const {organizationUrl} = links;
+
+                  const shouldUseLegacyRoute =
+                    !organizationUrl ||
+                    !organization.features.includes('customer-domains') ||
```

Comment:
Should we have a utility function for this style of check? I feel like I've seen it a few times now.

Reply:
@markstory I've delegated this to the `shouldUseLegacyRoute()` utility function.